pkg/crawl/identify.go

```diff
@@ -193,23 +193,24 @@ func (i *Log4jIdentifier) Identify(ctx context.Context, path string, d fs.DirEnt
 	if result != NothingDetected && len(versions) == 0 {
 		versions[UnknownVersion] = struct{}{}
 	}
-	return result, versions, err
+	return result, versions, skipped, err
 }
 
-func (i *Log4jIdentifier) findArchiveVulnerabilities(ctx context.Context, depth uint, walk archive.WalkFn, obfuscated bool) (Finding, Versions, error) {
+func (i *Log4jIdentifier) findArchiveVulnerabilities(ctx context.Context, depth uint, walk archive.WalkFn, obfuscated bool) (Finding, Versions, uint64, error) {
 	archiveResult := NothingDetected
 	versions := make(Versions)
+	var skipped uint64 = 0
 	i.Limiter.Take()
-	if err := walk(ctx, i.vulnerabilityFileWalkFunc(depth, &archiveResult, versions, obfuscated)); err != nil {
-		return NothingDetected, nil, err
+	if err := walk(ctx, i.vulnerabilityFileWalkFunc(depth, &archiveResult, versions, &skipped, obfuscated)); err != nil {
+		return NothingDetected, nil, 0, err
 	}
-	return archiveResult, versions, nil
+	return archiveResult, versions, skipped, nil
 }
 
-func (i *Log4jIdentifier) vulnerabilityFileWalkFunc(depth uint, result *Finding, versions Versions, obfuscated bool) archive.FileWalkFn {
+func (i *Log4jIdentifier) vulnerabilityFileWalkFunc(depth uint, result *Finding, versions Versions, skipped *uint64, obfuscated bool) archive.FileWalkFn {
 	return func(ctx context.Context, path string, size int64, contents io.Reader) (proceed bool, err error) {
 		archiveType, ok := i.ParseArchiveFormat(path)
-		if ok && depth < i.ArchiveMaxDepth {
+		if ok && depth < i.ArchiveMaxDepth && size < i.ArchiveMaxSize {
```

Comment:
@hpryce was this check against max archive size meant to be put back in?
This will mean block large tar archives from being traversed, even though they should not have worrying impact on memory.